src/App/LoadFigure.purs

```diff
@@ -55,22 +53,36 @@ loadSpec filename = do
       Left err -> error ("Json fetching failed with " <> printError err)
       Right response -> pure $ response.body
 
-loadFigureFromJson :: Json -> Effect Unit
-loadFigureFromJson json = runAffs_ (uncurry drawFig)
-   [ case decodeJson json :: Either JsonDecodeError JsonSpec of
-        Left err -> error ("JSON decoding failed with " <> show err)
-        Right spec -> ("fig" × _) <$> runWebT (FileCxt { fluidSrcPaths }) (loadFig figSpec)
-           where
-           figSpec@{ fluidSrcPaths } = figSpecFromJson spec
-   ]
+--loadFigure -- specFilename (String) & srcFilename (String)
+loadFigure :: String -> String -> Effect Unit
+loadFigure specFile srcFile = launchAff_ do
+   jsonSpec <- loadSpec specFile
+   liftEffect $ loadFigureSpec jsonSpec srcFile
+
+--loadFigureSrc -- specFilename (String) & fluidSrc (String)
+loadFigureSrc :: String -> String -> Effect Unit
+loadFigureSrc specFile fluidSrc = launchAff_ do
+   jsonSpec <- loadSpec specFile
+   liftEffect $ loadFigureSpecSrc jsonSpec fluidSrc
 
-loadFigure :: String -> Effect Unit
-loadFigure filename = launchAff_ do
-   jsonSpec <- loadSpec filename
-   liftEffect $ loadFigureFromJson jsonSpec
+--loadFigureSpec -- jsonSpec (Json) & srcFilename (String)
+loadFigureSpec :: Json -> String -> Effect Unit
+loadFigureSpec jsonSpec srcFile = launchAff_ do
+   fluidSrc <- loadFileFromPath (File (srcFile <> fluidExtension))
+   liftEffect $ loadFigureSpecSrc jsonSpec (definitely' fluidSrc)
 
-drawCode :: String -> String -> Effect Unit
-drawCode folder file = runAffs_ drawFile
-   [ runWebT (FileCxt { fluidSrcPaths: [ Folder folder ] }) $ loadFile_ [ Folder folder ] (File file)
+--loadFigureSpecSrc -- jsonSpec (Json) & fluidSrc (String)
+loadFigureSpecSrc :: Json -> String -> Effect Unit
+loadFigureSpecSrc jsonSpec fluidSrc = runAffs_ (uncurry drawFig)
+   [ case decodeJson jsonSpec :: Either JsonDecodeError JsonSpec of
+        Left err -> error ("JSON decoding failed with " <> show err)
+        Right spec -> do
+           let figSpec@{ fluidSrcPaths } = figSpecFromJson spec
+           ("fig" × _) <$> runWebT (FileCxt { fluidSrcPaths }) (loadFig figSpec fluidSrc)
    ]
 
+drawCode :: String -> Effect Unit
+drawCode file = launchAff_ do
+   fluidSrc <- loadFileFromPath (File (file <> fluidExtension))
+   liftEffect $ drawFile ((File file) × (definitely' fluidSrc))
+
```

Comment:
Spurious newline here -- you can set `files.trimFinalNewlines` in VSCode to remove these automatically.